test/controllers/users/registrations_controller_test.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module Users
+  class RegistrationsControllerTest < ActionDispatch::IntegrationTest
+    test "should register new user" do
+      user_payload = {
+        name: "name",
+        surname: "surname",
+        email: "test@example.com",
+        password: "password",
+        password_confirmation: "password"
+      }
+
+      assert_changes -> { User.find_by(email: user_payload[:email]) }, "User not added" do
+        post "/user", params: {
+          user: user_payload
+        }
+      end
+
+      new_user = User.find_by(email: user_payload[:email])
+
+      assert new_user.name == user_payload[:name], "Name not same as in payload"
```

Comment:
I'd better use `assert_equal` here
```suggestion
      assert_equal "name", new_user.name, "Name not same as in payload"
```

https://guides.rubyonrails.org/testing.html#available-assertions